jobsdb/jobsdb.go

```diff
@@ -3105,6 +3105,7 @@ func (jd *HandleT) getUnprocessed(ctx context.Context, params GetQueryParamsT) (
 	}
 	defer jd.dsListLock.RUnlock()
 
+	dsRangeList := jd.getDSRangeList()
```

Comment:
In line 1678 we might need to call `refreshDSRangeList` instead of `refreshDSList` otherwise there is a possibility to ignore the latest dataset for a brief period of time.

Reply:
imo we wouldn't be ignoring a dataset, we'd just not be applying

```
if *params.AfterJobID > dsRangeList[idx].maxJobID {
	continue
}
```

condition for the penultimate dataset which means we'd query it even if above condition was true.

But I do see your point, shouldn't we be changing `refreshDSList` to `refreshDSRangeList` in multiple places then?

Reply:
I don't believe it is needed elswhere, but let's do this change in another PR

Reply:
yes, another pr :)